src/main/java/core/basesyntax/SalaryInfo.java

```diff
@@ -1,7 +1,38 @@
 package core.basesyntax;
 
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+
 public class SalaryInfo {
+    private static final DateTimeFormatter FORMATTER = DateTimeFormatter.ofPattern("d.MM.yyyy");
+    private static final String HYPHEN_WITH_SPACES = " - ";
+    private static final String REPORT_START = "Report for period ";
+
     public String getSalaryInfo(String[] names, String[] data, String dateFrom, String dateTo) {
-        return null;
+        LocalDate fromDate = LocalDate.parse(dateFrom, FORMATTER).minusDays(1);
+        LocalDate toDate = LocalDate.parse(dateTo, FORMATTER).plusDays(1);
+        LocalDate dateFromData;
+        int[] salaries = new int[names.length];
+        String[] splitData;
+        StringBuilder stringBuilder = new StringBuilder()
+                .append(REPORT_START).append(dateFrom)
+                .append(HYPHEN_WITH_SPACES).append(dateTo);
+
+        for (int i = 0; i < names.length; i++) {
+
+            for (String lineOfData : data) {
+                splitData = lineOfData.split(" ");
```

Comment:
It would be more accurate from my perspective to make this literal a constant variable